### ISSUE TITLE
View change conductor test

### ISSUE THREAD

**Issue by guruhubb** [0chain/0chain#192]
There are view change test cases that have errors that need to be looked at -

view-change-1.yaml

start set Outside miner comes up in phase X
[Err] Test case 0: Outside miner comes up in phase 'start'
[OK]  Test case 1: Outside miner comes up in phase 'contribute'
[OK]  Test case 2: Outside miner comes up in phase 'share'
[OK]  Test case 3: Outside miner comes up in phase 'publish'
[OK]  Test case 4: Outside miner comes up in phase 'wait'

start set Miner goes down in phase X and doesn't come up in the same view change
[OK]  Test case 0: Miner goes down in phase 'start' and doesn't come up
[OK]  Test case 1: Miner goes down in phase 'contribute' and doesn't come up
[OK]  Test case 2: Miner goes down in phase 'share' and doesn't come up
[OK]  Test case 3: Miner goes down in phase 'publish' and doesn't come up
[OK]  Test case 4: Miner goes down in phase 'wait' and doesn't come up

start set Outside miner goes down in phase X and doesn't come up in the same view change
[ERR] Test case 0: Outside miner goes down in phase 'start' and doesn't come up
[ERR] Test case 1: Outside miner goes down in phase 'contribute' and doesn't come up
[ERR] Test case 2: Outside miner goes down in phase 'share' and doesn't come up
[ERR] Test case 3: Outside miner goes down in phase 'publish' and doesn't come up

start set Miner goes down and up
[OK]  Test case 0: Miner goes down in phase 'start' and comes up in phase 'start'
[OK]  Test case 1: Miner goes down in phase 'start' (already registered) and comes up in phase 'contribute'
[OK]  Test case 2: Miner goes down in phase 'contribute' (after sending mpk) and comes up in phase 'share'
[OK]  Test case 3: Miner goes down in phase 'publish' (after publishing signOrShares) and comes up in next VC

start set Artificially force X miner to send bad share to Y miners
[ERR] Test case 0: Artificially force 1 miner to send bad share to 1 miners
[ERR] Test case 1: Artificially force 1 miner to send bad share to 2 miners
[ERR] Test case 2: Artificially force 1 miner to send bad share to 3 miners
[ERR] Test case 3: Artificially force 2 miner to send bad share to 2 miners

start set Artificially force X miner to send share to Y miners only
[OK]  Test case 0: Artificially force 1 miner to send share to 2 miners only
[OK]  Test case 1: Artificially force 1 miner to send share to 1 miners only

start set Down and up on VC rounds
[OK]  Test case 0: Miner goes down on 497 round, and comes up shortly
[OK]  Test case 1: Two mines goes down on 497 round, and comes up shortly

start set Revealed
[OK]  Test case 0: Revealed

----
view-change-2.yaml

start set MPK
[OK]  Test case 1: Send no MPK to blockchain
[OK]  Test case 0: Send wrong MPK to blockchain
[OK]  Test case 2: Group send no MPK to blockchain

start set Shares
[OK]  Test case 0: Send wrong share
[OK]  Test case 1: Only send correct share to the group
[OK]  Test case 2: Send no shares

start set Signatures
[OK]  Test case 0: Don't send a signature
[ERR] Test case 1: Only send correct signature to the group
[OK]  Test case 2: Send no signatures

start set Publish
[OK]  Test case 0: Publish: Don't send a signature
[OK]  Test case 1: Publish: Only send correct signature to the group
[OK]  Test case 2: Publish: Send no signatures

----
view-change-3.yaml

start set Miner down/up
[OK]  Test case 0: Miner: 50 (switch to contribute)
[OK]  Test case 1: Miner: 100 (switch to share)
[OK]  Test case 2: Miner: 150 (switch to publish)
[ERR] Test case 3: Miner: 200 (switch to wait)
[ERR] Test case 4: Miner: vc-1 round
[OK]  Test case 5: Miner: vc round
[OK]  Test case 6: Miner: vc+2 rounds

start set Sharder down/up
[ERR] Test case 0: Sharder: 50 (switch to contribute)
[OK]  Test case 1: Sharder: 100 (switch to share)
[OK]  Test case 2: Sharder: 150 (switch to publish)
[OK]  Test case 3: Sharder: 200 (switch to wait)
[OK]  Test case 4: Sharder: vc-1 round
[OK]  Test case 5: Sharder: vc round
[OK]  Test case 6: Sharder: vc+2 rounds

start set All miners down/up
[ERR] Test case 0: All miners: 50 (switch to contribute)
[ERR] Test case 1: All miners: 100 (switch to share)
[ERR] Test case 2: All miners: 150 (switch to publish)
[ERR] Test case 3: All miners: 200 (switch to wait)
[OK]  Test case 4: All miners: vc-1 round
[ERR] Test case 5: All miners: vc round
[ERR] Test case 6: All miners: vc+2 rounds

start set All sharders down/up
[ERR] Test case 0: All sharders: 50 (switch to contribute)
[ERR] Test case 1: All sharders: 100 (switch to share)
[OK]  Test case 2: All sharders: 150 (switch to publish)
[ERR] Test case 3: All sharders: 200 (switch to wait)
[ERR] Test case 4: All sharders: vc-1 round
[ERR] Test case 5: All sharders: vc round
[ERR] Test case 6: All sharders: vc+2 rounds

start set All nodes down/up
[OK]  Test case 0: All nodes: 50 (switch to contribute)
[OK]  Test case 1: All nodes: 100 (switch to share)
[OK]  Test case 2: All nodes: 150 (switch to publish)
[ERR] Test case 3: All nodes: 200 (switch to wait)
[OK]  Test case 4: All nodes: vc-1 round
[OK]  Test case 5: All nodes: vc round

[2021-04-17.txt](https://github.com/0chain/0chain/files/6332180/2021-04-17.txt)

**Comment by bbist** [comment]
As view-change-1.yaml and view-change-3.yaml both deal with fault tolerance, they should be merged into view-change-fault-tolerance.yaml and more granular test cases should be added.
View-change-2.yaml deals with byzantine conditions, hence all byzantine test cases from view-change-1.yaml should be moved to view-change-2.yaml and more byzantine test cases should be added to increase coverage.

**Comment by guruhubb** [comment]
Ok sure go ahead @bbist

**Comment by bbist** [comment]
This will be addressed once PR #283 is merged.

**Comment by Sriep** [comment]
Closed as PR283 has been merged.
